### DISCUSSION TITLE
Abnormal Linux data collected by MAL

### DISCUSSION THREAD

**Discussion by brave-programmer** [apache/skywalking#11427] [Q&A]
otel-rules/vm.yaml config:
  - name: cpu_total_percentage
    exp: (node_cpu_seconds_total * 100).tagNotEqual('mode' , 'idle').sum(['node_identifier_host_name']).rate('PT1M')
  - name: cpu_total_percentage_idle
    exp: (node_cpu_seconds_total * 100).tagEqual('mode' , 'idle').sum(['node_identifier_host_name']).rate('PT1M')

dashboard show:
cpu_total_percentage   percentage is ok.
cpu_total_percentage_idle  percentage is too big???

**Comment by wu-sheng** [comment]
Could you write in a complete sentence? `cpu_total_percentage_idle percentage is too big???` this doesn't make sense. Hard to understand.

**Reply by brave-programmer** [reply]
A two core Linux, theoretically, the sum of idle and non idle should be equal to 200%, but it shows 30% non idle and 2328954% idle

**Reply by wu-sheng** [reply]
I don't think we have `cpu_total_percentage_idle`. If you are writing the MAL, you should check the RAW data reported by OTEL Linux exporter.
We don't have your data or env, it is hard to say.

**Reply by brave-programmer** [reply]
I have tried multiple Linux platforms and the results are all abnormal. But it is normal to use PromQL statistics in Prometheus. 
you can try it. add config on otel-rules/vm.yaml. and show on dashboard(Can monitor any LINUX). 
- name: cpu_total_percentage_idle
   exp: (node_cpu_seconds_total * 100).tagEqual('mode' , 'idle').sum(['node_identifier_host_name']).rate('PT1M')

**Reply by wu-sheng** [reply]
I would not try. Sorry. We have thousands of metrics reported and processed through various features. 
I am unaffordable to test one by one, especially to test something we don't declare about.

**Reply by brave-programmer** [reply]
ok, thanks!
